server/src/main/java/org/elasticsearch/transport/RemoteClusterConnection.java

```diff
@@ -125,7 +125,7 @@ void collectNodes(ActionListener<Function<String, DiscoveryNode>> listener) {
                     transportService.sendRequest(
                         connection,
                         RemoteClusterNodesAction.NAME,
-                        RemoteClusterNodesAction.Request.INSTANCE,
+                        new RemoteClusterNodesAction.Request(),
```

Comment:
This is the centre piece of the fix.

Reply:
Having sigletons is better than having to fill the boolean argument. Will make the change accordingly.